src/main/java/com/bio4j/angulillos/TypedVertex.java

```diff
@@ -10,27 +10,31 @@
   A typed vertex. A vertex and its type need to be defined at the same time. The vertex keeps a reference of its type, while the type works as a factory for creating vertices with that type.
 */
 interface TypedVertex <
-  N extends TypedVertex<N,NT,G,RV>,
-  NT extends TypedVertex.Type<N,NT,G,RV>,
-  G extends TypedGraph<G,RV,?>,
-  RV
+  N   extends TypedVertex<N,NT,G,RV,RE>,
+  NT  extends TypedVertex.Type<N,NT,G,RV,RE>,
+  G   extends TypedGraph<G,RV,RE>,
+  RV,
+  RE // NOTE we need RE if we want to safely add edges
```

Comment:
I don't want. I think that having typesafe `addVertex`, `addEdge` in the `TypedGraph` is more than enough. They are completely non-essential here and this parameter is needed _only_ for them.